server/config/config.go

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright (c) 2018 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/heketi/heketi/apps/glusterfs"
+	"github.com/heketi/heketi/middleware"
+)
+
+type Config struct {
+	Port                 string                   `json:"port"`
+	AuthEnabled          bool                     `json:"use_auth"`
+	JwtConfig            middleware.JwtAuthConfig `json:"jwt"`
+	BackupDbToKubeSecret bool                     `json:"backup_db_to_kube_secret"`
+	EnableTls            bool                     `json:"enable_tls"`
+	CertFile             string                   `json:"cert_file"`
+	KeyFile              string                   `json:"key_file"`
+
+	// pull in the config sub-object for glusterfs app
+	GlusterFS *glusterfs.GlusterFSConfig `json:"glusterfs"`
```

Comment:
Why is this hunk not added when this module is introduced, but in the patch "server: consolidate...." ?

Reply:
I was trying to refactor one area at a time. So the first patch touched on stuff that was in main.go and then later patches brings in the stuff from apps/glusterfs. Do you think there's a better approach than that?

Reply:
Ok, that makes sense... :-)